cli/azd/internal/cmd/add/add_select_ai.go

```diff
@@ -0,0 +1,192 @@
+package add
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+
+	armruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/azure/azure-dev/cli/azd/pkg/azureutil"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+)
+
+func (a *AddAction) selectOpenAi(console input.Console, ctx promptCtx) (r *project.ResourceConfig, err error) {
+	resourceToAdd := &project.ResourceConfig{}
+	aiOption, err := console.Select(ctx, input.ConsoleOptions{
+		Message: "Which type of Azure OpenAI service?",
+		Options: []string{
+			"Chat (GPT)",                   // 0 - chat
+			"Embeddings (Document search)", // 1 - embeddings
+		}})
+	if err != nil {
+		return nil, err
+	}
+
+	resourceToAdd.Type = project.ResourceTypeOpenAiModel
+
+	var allModels []ModelList
+	for {
+		err = provisioning.EnsureSubscriptionAndLocation(ctx, a.envManager, a.env, a.prompter, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		cred, err := a.creds.CredentialForSubscription(ctx, a.env.GetSubscriptionId())
+		if err != nil {
+			return nil, fmt.Errorf("getting credentials: %w", err)
+		}
+
+		pipeline, err := armruntime.NewPipeline(
```

Comment:
Guessing there's no wrapper in `azure-sdk-for-go` for this part of the management plane and so we had to do our own thing?

Reply:
Indeed, it's a pretty undocumented API right now